project-addons/custom_partner/partner.py

```diff
@@ -272,16 +272,21 @@ def create(self, vals):
 
     @api.multi
     @api.constrains('web')
-    def write(self, vals):
+    def check_client_type(self, vals):
```

Comment:
Los api.constrains no tienen vals como parámetro, sólo self.
Entonces la comprobación sería: "if not self.web and self.prospective:" y las excepciones de lanzan como ValidationError